lucene/core/src/java/org/apache/lucene/search/FilteringFieldComparator.java

```diff
@@ -68,10 +68,12 @@ public int compareValues(T first, T second) {
    * @param comparator – comparator to wrap
    * @param reverse – if this sort is reverse
    * @param singleSort – true if this sort is based on a single field and there are no other sort fields for tie breaking
+   * @param hasAfter – true if this sort has after FieldDoc
    * @return comparator wrapped as a filtering comparator or the original comparator if the filtering functionality
    * is not implemented for it
    */
-  public static FieldComparator<?> wrapToFilteringComparator(FieldComparator<?> comparator, boolean reverse, boolean singleSort) {
+  public static FieldComparator<?> wrapToFilteringComparator(FieldComparator<?> comparator, boolean reverse, boolean singleSort,
+      boolean hasAfter) {
```

Comment:
Do we really need to add the `hasAfter` ? Can we check the if the `topValue` in the DocComparator is greater than 0 instead ?

Reply:
At that moment topValue is not set yet, it will be set later in the constructor of `PagingFieldCollector`.